### PR TITLE
Bitwise ops improvements

### PR DESCRIPTION
**Pull request by zasdfgbnm**
- Bitwise shift remove floating point support
- Bitwise and, or, xor add (scalar, tensor) overload
- Use `test_ops.py` to test these ops, including error cases

### CODE REVIEW

File: test/test_binary_ufuncs.py [hunk 1]
```diff
@@ -3353,7 +3353,7 @@ def test_bitwise_xor(self, device):
             ),
         )
 
-    @dtypes(torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64)
+    @dtypes(torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64, torch.float)
```
Comment (ngimel) [inline test/test_binary_ufuncs.py:3356]: can you add a todo to port this test to error_inputs in OpInfos?

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
QUantization, seriously?

File: test/test_binary_ufuncs.py (hunk 1 above)
Comment (zasdfgbnm) [reply test/test_binary_ufuncs.py]: I updated this PR and removed this test

Comment (ngimel) [reply test/test_binary_ufuncs.py]: Awesome, thanks!

**Comment by Lezcano** [comment]
Could you also update the kernels (there is a dedicated path that implements this) and the docs?
I just found last week about this weird behaviour and it is even documented.

**Comment by zasdfgbnm** [comment]
@Lezcano which kernel are you referring to? Are you talking about the kernel that I already removed from my previous PR? https://github.com/pytorch/pytorch/pull/77146/files#diff-0fd0c99904c62a1a7e46b08568d047c4e9ab153b529a8c6a2317ac6a3726bdff

**Comment by Lezcano** [comment]
oops. Yes, I was referring to that kernel, thanks!
Also, note that, at the moment, for `a << b`, if `b` is greater than `sizeof(a)` this is UB. We should use a ternary operator there really. Same in CUDA.
For the r-shift, I'm not sure whether `a >> b` is well-defined when `b > sizeof(a)`.

**Comment by zasdfgbnm** [comment]
I think starting from C++20, they are no longer undefined https://en.cppreference.com/w/cpp/language/operator_arithmetic. I guess PyTorch should not worry about whatever C++ does not provide in this case? When C++20 is adopted, this issue will be automatically resolved.

**Comment by ngimel** [comment]
Yeah, we are doing whatever c++ does (and that includes implementation-defined logical/binary rshift for signed args)

**Pull request closed without merging**